scripts/deploy.sh

```diff
@@ -18,55 +18,61 @@ else
     exit 0
 fi
 
-function update_latest {
-    local out_path=$1
+function move_sample_scripts {
+    local subdirectory=$1
+    for f in $(find ./samples/$subdirectory -name '*.html'); do
+       sed -i -E "s/((\.\.\/)+dist\/)/..\/\1$subdirectory\//" $f
+    done
+}
+
+function update_with_tag {
+    local tag=$1
+    rm -rf "$tag/docs"
+    cp -r ../dist/docs $tag
+    rm -rf "$tag/samples"
+    cp -r ../dist/samples $tag
 
-    # The most recent directory whose name is a version number
-    local latest=($(ls -v $out_path | egrep '^('$VERSION_REGEX')$' | tail -1))
+    move_sample_scripts $tag
 
-    # As soon as a single version is deployed this line is a no-op
-    if [ "$latest" == "" ]; then latest='master'; fi
+    deploy_versioned_files $tag
+}
 
-    # Don't update "latest" on alpha or beta releases
-    if [[ "$latest" =~ ^[^-]+$ ]]; then
-        rm -f $out_path/latest
-        ln -s $latest $out_path/latest
+# https://www.chartjs.org/docs/next|latest/
+# https://www.chartjs.org/samples/next|latest/
+# https://www.chartjs.org/dist/next|latest/Chart.*js
+function update_tagged_files {
+    if [ "$VERSION" == "master" ]; then
+        return
     fi
 
-    # Always update "next"
-    rm -f $out_path/next
-    ln -s $latest $out_path/next
+    # Don't update "latest" on alpha or beta releases but update "next"
+    if [[ "$VERSION" =~ ^[^-]+$ ]]; then
+        update_with_tag lastest
+    else
+        update_with_tag next
+    fi
 }
 
-function deploy_files {
-    local in_files=$1
-    local out_path=$2
-    rm -rf $out_path/$VERSION
-    mkdir -p $out_path/$VERSION
-    cp -r $in_files $out_path/$VERSION
-    update_latest $out_path
+function deploy_versioned_files {
+    local version=$1
```

Comment:
could use `$VERSION` here to be consistent with `update_tagged_files` (or the other way)

Reply:
I'm not sure I understand this suggestion. `$VERSION` is a global variable. I'm using `$version` here to avoid colliding with it